lib/iglu-client/resolver.rb

```diff
@@ -50,14 +52,24 @@ def lookup_schema(schema_key)
         if lookup_result.nil?
           raise Registries::ResolverError.new(failures, schema_key)
         else
-          @cache[schema_key] = lookup_result
+          store_time = Time.now.getutc
+          @cache[schema_key] = [lookup_result, store_time]
           lookup_result
         end
       else
         if cache_result.is_a?(Registries::ResolverError)
           raise cache_result
         else
-          cache_result
+          if cache_result.is_a? (Array)
+              store_time = cache_result[1]
+              time_diff = (fetch_time - store_time).round
+              if time_diff >= @cacheTtl
+                 @cache.delete(schema_key)
```

Comment:
We need not just delete cached item, but replace it with new.

Whole point of cache is that your cached item *can* be outdated after TTL and therefore you need to *refresh* it. In this implementation it is *always* outdated after TTL.